airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -121,29 +122,69 @@ async def _run(self) -> StepResult:
                     ]
                 )
             )
-            # Set GRADLE_HOME and GRADLE_USER_HOME to the directory which will be rsync-ed with the gradle cache volume.
+            # Set GRADLE_HOME to the directory which will be rsync-ed with the gradle cache volume.
             .with_env_variable("GRADLE_HOME", "/root/.gradle")
+            # Same for GRADLE_USER_HOME.
             .with_env_variable("GRADLE_USER_HOME", "/root/.gradle")
             # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
             # This is consumed by settings.gradle in the repo root.
             .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Disable the Ryuk container because it needs privileged docker access which it can't have.
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true")
+            # Set the current working directory.
+            .with_workdir("/airbyte")
+        )
+
+        # Mount the whole git repo to update the persistent gradle cache and build the CDK.
+        with_whole_git_repo = (
+            gradle_container_base
+            # Mount the whole repo.
+            .with_mounted_directory("/airbyte", self.context.get_repo_dir("."))
```

Comment:
The last time I tried mounting the whole repo without `include` it was a very long operation when running locally, due to the number/size of files to mount to the dagger build context.
Can we only mount the "gradle projects"?

Reply:
In my experience it wasn't noticeably long, like a couple of seconds. I didn't look into it too deeply. In any case, yes, let's only mount directories which either have a build.gradle file or recursively contain a subdirectory which contains a build.gradle file. Is there a shorthand for this?

Reply:
I don't think a glob is going to cut it. Are you OK with me doing this change in a followup PR?

Reply:
Yep!